pkg/simple/client/alerting/options.go

```diff
@@ -26,13 +39,37 @@ func NewAlertingOptions() *Options {
 	}
 }
 
-func (s *Options) ApplyTo(options *Options) {
-	if options == nil {
-		options = s
-		return
-	}
+func (o *Options) ApplyTo(options *Options) {
+	reflectutils.Override(options, o)
+}
 
-	if s.Endpoint != "" {
-		options.Endpoint = s.Endpoint
+func (o *Options) Validate() []error {
+	errs := []error{}
+
+	if len(o.ThanosRuleResourceLabels) > 0 {
+		lblStrings := strings.Split(o.ThanosRuleResourceLabels, ",")
+		for _, lblString := range lblStrings {
+			if len(lblString) > 0 {
+				lbl := strings.Split(lblString, "=")
+				if len(lbl) != 2 {
+					errs = append(errs, fmt.Errorf("invalid alerting-thanos-rule-resource-labels arg: %s", o.ThanosRuleResourceLabels))
+					break
+				}
+			}
+		}
 	}
+
+	return errs
+}
+
+func (o *Options) AddFlags(fs *pflag.FlagSet, c *Options) {
+	fs.StringVar(&o.Endpoint, "alerting-server-endpoint", c.Endpoint,
+		"alerting server endpoint for the alerting with v1 version.")
+
+	fs.StringVar(&o.PrometheusEndpoint, "alerting-prometheus-endpoint", c.PrometheusEndpoint,
+		"Prometheus service endpoint from which built-in alerting rules are gotten(for the alerting with v2alpha1 version or higher versions)")
+	fs.StringVar(&o.ThanosRulerEndpoint, "alerting-thanos-ruler-endpoint", c.ThanosRulerEndpoint,
+		"Thanos ruler service endpoint from which custom alerting rules are gotten(for the alerting with v2alpha1 version or higher versions)")
+	fs.StringVar(&o.ThanosRuleResourceLabels, "alerting-thanos-rule-resource-labels", c.ThanosRuleResourceLabels,
+		"The labels will be added to prometheusrule custom resources to be selected by thanos ruler. eg: thanosruler=thanos-ruler,role=custom-alerting-rules (for the alerting with v2alpha1 version or higher versions)")
```

Comment:
The labels will be added to prometheusrule custom resources to be selected by thanos ruler =>
Labels used by Thanos Ruler to select PrometheusRule custom resources